### PR TITLE
fix(revert 27883): Excess padding in horizontal Bar charts

### LINKED ISSUE #29342: Excess Padding in Horizontal Bar Charts When Legend is Added
### Bug description

Horizontal bar charts in Apache Superset are created with extra padding when a legend is added, affecting the chart's layout and appearance. This issue was introduced in [PR #27883](https://github.com/apache/superset/pull/27883#issuecomment-2182972005).

**Expected Behavior:**
The chart should adjust the layout to accommodate the legend without adding excessive padding.

### How to reproduce the bug

1. Create a horizontal bar chart
2. Ensure that the legend is enabled and the margin is set to 0 or none
3. You should see extra padding between the chart and the legend

### Screenshots/recordings

image

### Superset version

4.0.1

### Python version

3.9

### Node version

16

### Browser

Chrome

### Additional context

This bug was validated by @sfirke in Slack

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [x] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
This PR reverts https://github.com/apache/superset/pull/27883 which incorrectly calculated the chart's padding and introduced https://github.com/apache/superset/issues/29342.

There's a [bug](https://github.com/apache/echarts/issues/20068) with `yAxis.inverse` in ECharts and we don't know when a fix will be provided. To fix this issue, this PR partially reverts https://github.com/apache/superset/pull/23273 which introduced said configuration. I'm purposefully not adding a migration to this PR, as https://github.com/apache/superset/pull/23273 did, so it can be cherry-picked as it's affecting all 4.0.x versions. This will change the default y-axis order of horizontal bar charts. @yousoph 

An alternative solution, would be to fix the ECharts bug. @villebro 

Fixes https://github.com/apache/superset/issues/29342

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screenshot 2024-06-24 at 11 16 37
Screenshot 2024-06-24 at 11 16 02
Screenshot 2024-06-24 at 10 51 52
Screenshot 2024-06-24 at 10 51 27

### TESTING INSTRUCTIONS
1 - Create an horizontal bar chart using a time column in the x-axis
2 - Check that the chart is displayed correctly.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by rusackas — APPROVED** [review]
Approving to unblock, but wondering if the fix-forward is simply changing a number.

File: superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/transformers.ts
Comment (michael-s-molina) [reply superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/transformers.ts:576]: You cannot set a fixed padding because it will depend on the size of each bar generated by ECharts, which will also vary depending on the number of bars. This should be done automatically by ECharts.

**Comment by michael-s-molina** [comment]
> Approving to unblock, but wondering if the fix-forward is simply changing a number.

Before merging this, I still want @villebro's input. Maybe we can fix the bug in ECharts and avoid reverting https://github.com/apache/superset/pull/23273.

**Review by villebro — APPROVED** [review]
LGTM

**Pull request merged**
